docs: architecture diagram

aliziel opened the pull request:
Addresses #87 

Needs some fresh eyes: Is it easy to follow ? Is there too much/too little information ? Don't be shy with feedback !

## Checklist

<!-- Feel free to remove any items that don't apply to your PR (e.g. small fixes may not require documentation updates). -->

- [ ] Pre-commit hooks pass (`uv run prek run --all-files`)
- [x] Documentation updated
- [ ] Tests pass (`uv run pytest --integration`)
- [ ] The PR's title is formatted per [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)

gadomski reviewed (commented):
I think it looks great! My only note is that the load balancer diagram makes everything look more complicated than it really is. How hard would it be to make two versions, one with only one AZ displayed and one with all three?

aliziel commented:
> How hard would it be to make two versions, one with only one AZ displayed and one with all three?

@gadomski Not hard ! And totally agree. 

My concern with doing 1 + 3 AZ versions is it could be unclear whats actually live and whats in triplicate if separated, which could mislead both owners and maintainers.

I just pushed a couple simplified options and will go for the 1 AZ + 3 AZ idea if neither convey what we need:

1. Revised Original (TIL: onion skin [diff](https://github.com/AdaptationAtlas/adaptation-atlas-assistant/pull/162/commits/6081808d61336c390cd9299dbef7419baa9484f4))
    1. Removed the AZ-level capacity nodes
    1. Moved some services outside of the VPC
        1. This was for accuracy, as well as readability. They were originally represented within the VPC, but they're just secured resources in the account
    1. Consolidated supporting services to one area to give resource foundation more space
1. Minimalist ([docs/simple.svg](https://github.com/AdaptationAtlas/adaptation-atlas-assistant/pull/162/commits/dde330310dabb9687667d055e91dd306e493d387))
    1. Only include runtime infra
    1. Collapse second and third AZs
    1. Remove descriptions

camillecroft commented:
> > In step 5 you address secrets storage. Would it be beneficial to add that the GitHub Actions secrets and variables are used to store configuration/credentials?
> 
> @camillecroft What about something like [efe1577](https://github.com/AdaptationAtlas/adaptation-atlas-assistant/commit/efe1577e045845f0ed4fa1fd8cc25c5bdc8e1606), essentially:
> 
> ```diff
> - Application secrets are stored in Secrets Manager for reference by containers.
> + Application secrets are sourced from GitHub and Terraform and stored in Secrets Manager for reference by containers. 
> Sensitive values are not written into task definitions.
> ```
> 
> Added Terraform because the Cognito vars are using the ARN output from build. Still holding out hope for a more elegant secrets solution, but greater detail will be in docs for sure (in-progress).

Sounds good, great thanks!
